src/rest.ts

```diff
@@ -156,4 +188,13 @@ export const rest = {
   delete: createRestHandler(RESTMethods.DELETE),
   patch: createRestHandler(RESTMethods.PATCH),
   options: createRestHandler(RESTMethods.OPTIONS),
+  link: (rootURL: string) => ({
```

Comment:
Take a look at how `graphql.link` API is implemented:

https://github.com/mswjs/msw/blob/7585f5440fec589524626bfa70ec8ed666faf986/src/graphql.ts#L267

 It leverages a high-order function that produces the list of request handlers so you don't have to repeat them. I highly suggest to adopt a similar approach. It would be almost the same to what you have, with the exception of using a single `createRestHandler` function that accepts an optional root. In case of un-link'ed handlers it would be a permissive mask `*`, allowing any requests to be captured.

Reply:
Will look into that, thanks

Reply:
As for the root being optional, I do not think it's something that we should do. The idea here is to solve the issue where we pre-define the root, If someone wants to have the root as `*` let them specify that explicitly. WYD?

Reply:
What I meant is to use an internal creator function which has an optional root:

```js
function createRestHandlersWithBase(base: Mask) {
  return { get, post, put, ... }
}

// These are the default handlers like "rest.get", with a permissive base.
const defaultRestHandlers = createRestHandlersWithBase('*')

export const rest = {
  ...defaultRestHandlers,
  // Pass the function reference so that people can create their own scoped handlers
  // retaining the same return signature from the internal handler generator function.
  link: createRestHandlersWithBase
}
```